src/main/scala/org/broadinstitute/dsde/agora/server/webservice/util/DockerHubClient.scala

```diff
@@ -1,48 +1,35 @@
 package org.broadinstitute.dsde.agora.server.webservice.util
 
-import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
 import org.broadinstitute.dsde.agora.server.exceptions.DockerImageNotFoundException
-import spray.client.pipelining._
-import spray.http.{HttpResponse, StatusCodes}
 import org.broadinstitute.dsde.agora.server.webservice.util.DockerHubJsonSupport._
 
-import scala.concurrent.duration._
-import scala.concurrent.{Await, Future}
-import scala.util.{Failure, Success}
+import scala.concurrent.Future
 
 case class DockerImageReference(user: Option[String], repo: String, tag: String)
 
 case class DockerTagInfo(pk: Int, id: String)
 
-object DockerHubClient {
+trait DockerHubClient extends HttpClient {
 
-  implicit val actorSystem = ActorSystem("agora")
-
-  import actorSystem.dispatcher
-
-  val timeout = 20.seconds
-
-  def pipeline = sendReceive
-
-  def dcGET(url: String): Future[HttpResponse] = pipeline(Get(url))
+  def dcGET(url: String): Future[HttpResponse] = get(Uri(url))
 
   def doesDockerImageExist(dockerImage: DockerImageReference): Future[Boolean] = {
     val dockerImageInfo: Future[HttpResponse] = dcGET(dockerImageTagUrl(dockerImage))
 
-    dockerImageInfo transform(
-      { response: HttpResponse => //on success
+    dockerImageInfo flatMap { response: HttpResponse => //on success
         response.status match {
-          case StatusCodes.OK => unmarshal[List[DockerTagInfo]].apply(response).nonEmpty
+          case StatusCodes.OK => Unmarshal(response.entity).to[List[DockerTagInfo]] map (_.nonEmpty)
           case StatusCodes.NotFound => throw DockerImageNotFoundException(dockerImage)
           case _ => throw DockerImageNotFoundException(dockerImage)
         }
-      }, { exc => //on failure
-        throw DockerImageNotFoundException(dockerImage)
+      } recoverWith { //on failure
+        case e => throw DockerImageNotFoundException(dockerImage)
       }
-    )
   }
 
-  val dockerImageRepositoryBaseUrl = "https://index.docker.io/v1/repositories/"
```

Comment:
Consider replacing lines 32-40 with

```scala
  val dockerImageRepositoryBaseUrl = "https://index.docker.io/v1/repositories/"

  def dockerImageTagUrl(dockerImage: DockerImageReference) = {
    val maybeUser = dockerImage.user.map(_ + "/").getOrElse("")

    dockerImageRepositoryBaseUrl + maybeUser + dockerImage.repo + "/tags/" + dockerImage.tag
  }
```

It passes the unit tests with 

```scala
override val dockerImageRepositoryBaseUrl = s"http://localhost:$mockServerPort/"
```
in `DockerHubClientSpec`